File: src/#589/solution.go
package solution

type Node struct {
	Val      int
	Children []*Node
}

func preorder(root *Node) []int {
	var arr []int
	helper(root,&arr)
	return arr
}

func helper(node *Node, arr *[]int) {
	if node == nil {
		return
	}
	*arr = append(*arr, node.Val)
	for _, n := range node.Children {
		helper(n, arr)
	}
}

File: src/#589/solution.js
function Node(val, children) {
  this.val = val
  this.children = children
}

var preorder = function (root) {
  var arr = []

  helper(root)

  return arr

  function helper(node) {
    if (node === null) {
      return
    }
    arr.push(node.val)
    for (let i = 0; i < node.children.length; i++) {
      helper(node.children[i])
    }
  }
}
